config-model/src/test/java/com/yahoo/searchdefinition/processing/ImportedFieldsTestCase.java

```diff
@@ -65,6 +69,173 @@ private static Search buildAdSearch(String sdContent) throws ParseException {
+            str_int_map_value_attr = true;
+        }
+
+        public ParentSdBuilder elem_array_name_attr(boolean v) { elem_array_name_attr = v; return this; }
+        public ParentSdBuilder elem_array_weight_attr(boolean v) { elem_array_weight_attr = v; return this; }
+        public ParentSdBuilder elem_map_key_attr(boolean v) { elem_map_key_attr = v; return this; }
+        public ParentSdBuilder elem_map_value_name_attr(boolean v) { elem_map_value_name_attr = v; return this; }
+        public ParentSdBuilder elem_map_value_weight_attr(boolean v) { elem_map_value_weight_attr = v; return this; }
+        public ParentSdBuilder str_int_map_key_attr(boolean v) { str_int_map_key_attr = v; return this; }
+        public ParentSdBuilder str_int_map_value_attr(boolean v) { str_int_map_value_attr = v; return this; }
+
+        public String build() {
+            return joinLines("search parent {",
+                    "  document parent {",
+                    "    struct elem {",
+                    "      field name type string {}",
+                    "      field weight type int {}",
+                    "    }",
+                    "    field elem_array type array<elem> {",
+                    "      indexing: summary",
+                    "      struct-field name {",
+                    structFieldSpec(elem_array_name_attr, true),
+                    "      }",
+                    "      struct-field weight {",
+                    structFieldSpec(elem_array_weight_attr, false),
+                    "      }",
+                    "    }",
+                    "    field elem_map type map<string, elem> {",
+                    "      indexing: summary",
+                    "      struct-field key {",
+                    structFieldSpec(elem_map_key_attr, true),
+                    "      }",
+                    "      struct-field value.name {",
+                    structFieldSpec(elem_map_value_name_attr, true),
+                    "      }",
+                    "      struct-field value.weight {",
+                    structFieldSpec(elem_map_value_weight_attr, false),
+                    "      }",
+                    "    }",
+                    "    field str_int_map type map<string, int> {",
+                    "      indexing: summary",
+                    "      struct-field key {",
+                    structFieldSpec(str_int_map_key_attr, true),
+                    "      }",
+                    "      struct-field value {",
+                    structFieldSpec(str_int_map_value_attr, false),
+                    "      }",
+                    "    }",
+                    "  }",
+                    "}");
+        }
+
+        private static String structFieldSpec(boolean isAttribute, boolean isFastSearch) {
+            List<String> result = new ArrayList<String>();
+            if (isAttribute) {
+                result.add("        indexing: attribute");
+                if (isFastSearch) {
```

Comment:
Is the fast-search aspect really needed / used in these tests? If not, consider removing it to simplify.